### PR TITLE
Handle no test runs better.

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Pulls the `.info` styles out into a reusable component, and surfaces another one for the case that we're finished loading but there's no test runs.

@mdittmer - this also re-jigs the code around the spinner (needed `isLoading` elsewhere) - let me know if the `newValue` / `oldValue` stuff was important in some way that I'm missing.

## Review Information
See autodeployed environment, load something with no results, e.g. /results?sha=1234567890

### CODE REVIEW

File: webapp/components/info-banner.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        float: right;
+      }
+      .banner.error {
+        background-color: var(--paper-red-100);
+        border-left: solid 4px var(--paper-red-300);
+      }
+    </style>
+
+    <section class$="banner [[type]]">
+      <slot></slot>
```
Comment (mdittmer) [inline webapp/components/info-banner.html]: What's this for? Does it denote where content prior to content inside `<... slot="small">...</...>` will go?

File: webapp/components/wpt-results.html [hunk 2]
```diff
@@ -173,56 +164,64 @@
...
-              <th><test-run test-run="[[testRun]]"></test-run></th>
-            </template>
-            <template is="dom-if" if="[[diffShown]]">
-              <th><test-run test-run="[[diffRun]]"></test-run></th>
-            </template>
-          </tr>
-        </thead>
-
-        <tbody>
+    <template is="dom-if" if="[[hasRuns]]">
```
Comment (mdittmer) [inline webapp/components/wpt-results.html]: This diff seems needlessly hard to follow. Did you just wrap existing markup in `<template is="dom-if" if="[[hasRuns]]">...</template>`, or is there more here?

File: webapp/components/wpt-results.html [hunk 3]
```diff
@@ -406,20 +412,12 @@ <h3>
...
-      observeLoadingCount(newValue, oldValue) {
-        if (newValue === oldValue) {
-          return;
-        }
-        const elem = this.shadowRoot.querySelector('paper-spinner-lite');
-        if (!elem) {
-          return;
-        }
+      computeIsLoading(loadingCount) {
+        return loadingCount;
```
Comment (mdittmer) [inline webapp/components/wpt-results.html]: nit: Explicit coercing to Boolean a bit clearer: `!!loadingCount`.

File: webapp/components/info-banner.html (hunk 1 above)
Comment (lukebjerring) [reply webapp/components/info-banner.html]: Correct - nameless is the default `slot`, which takes the (unslotted) child elements.
The other slot is named, so takes the element with `slot="small"`

https://www.polymer-project.org/2.0/docs/devguide/shadow-dom#shadow-dom-and-composition

File: webapp/components/wpt-results.html (hunk 2 above)
Comment (lukebjerring) [reply webapp/components/wpt-results.html]: Yep.

File: webapp/components/wpt-results.html (hunk 3 above)
Comment (lukebjerring) [reply webapp/components/wpt-results.html]: Done.

File: api/query/autocomplete.go [hunk 4]
```diff
@@ -123,7 +123,7 @@ func prepareAutocompleteResponse(limit int, filters *shared.QueryFilter, testRun
 		}
 	}
 
-	files := make([]AutocompleteResult, 0, fileSet.Cardinality()/len(testRuns))
+	var files []AutocompleteResult
```
Comment (Hexcles) [inline api/query/autocomplete.go]: Hmm, why is this changed?

Comment (lukebjerring) [reply api/query/autocomplete.go]: Drive-by fix for 500 when no test runs are found for an autocomplete query.

Comment (Hexcles) [reply api/query/autocomplete.go]: Hmm, sorry could you explain what was wrong with the original statement?

Comment (lukebjerring) [reply api/query/autocomplete.go]: `len(test runs)` is zero when no test runs get loaded. Division by zero is bad.

Comment (Hexcles) [reply api/query/autocomplete.go]: Ahh I see. Thanks!
